### PR TITLE
Gpii 2401

### PR DESCRIPTION
**Pull request by michelled**
Implemented the wait dialog when someone keys in or out.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -181,10 +223,65 @@ gpii.app.exit = function (that) {
...
+    }
+};
+
+/**
+ * Shows the dialog on users screen with the message passed as parameter.
+ * Records the time it was shown in `dialogStartTime` which we need when
+ * dismissing it (checking whether it's been displayed for the minimum amount of time)
+ *
+ * @param that {Object} the app module
+ * @param msg {String} the message to display to the user
```
Comment (kaspermarkus) [inline src/app.js]: remove this param

File: src/app.js [hunk 2]
```diff
@@ -181,10 +223,65 @@ gpii.app.exit = function (that) {
...
+    }, 100);
+
+    that.dialogStartTime = Date.now();
+};
+
+/**
+ * Dismisses the dialog. If less than `that.dialogMinDisplayTime` ms have passed since we first displayed
+ * the window, the function waits until `dialogMinDisplayTime` has passed before dismissing it.
+ *
+ * @param that {Object}: the app
```
Comment (kaspermarkus) [inline src/app.js]: no ":" after the brackets - according to our undocumented and unused parsing rules for the auto documentation

File: package.json [hunk 3]
```diff
@@ -26,7 +26,7 @@
     "electron": "1.4.1",
     "request": "2.69.0",
     "infusion": "git://github.com/amb26/infusion.git#240c00be023732ac8dcc193a57bc1b46c58d2de7",
-    "gpii-windows": "git://github.com/gpii/windows.git#hst-2017",
+    "gpii-windows": "git://github.com/kaspermarkus/windows.git#5fc0d9c65db1ac9cff9495f87bc36907d4ecd499",
```
Comment (kaspermarkus) [inline package.json]: This no longer references the latest (and merged) version of the universal GPII-2402 work.... Not sure which branch to best reference here - but I guess @stegru's https://github.com/GPII/windows/pull/122 branch (as soon as he updates the universal reference) would be the most sensible one

**Comment by kaspermarkus** [comment]
@amb26 or @javihernandez I'll be happy to test this one manually in terms of functionality, but since it partly includes my code, I can't be the main reviewer

**Comment by javihernandez** [comment]
@kaspermarkus, In fact I already started working towards testing this pull. First step was updating the reference of universal in windows and apparently there's some problem with it. I've just created a pull request (https://github.com/GPII/windows/pull/127) in order to check whether the problem also happens in CI.

File: src/app.js [hunk 4]
```diff
@@ -181,10 +223,65 @@ gpii.app.exit = function (that) {
...
 /**
-  * Handles when a user token is keyed out through other means besides the task tray key out feature.
-  * @param that {Component} An instance of gpii.app
-  * @param keyedOutUserToken {String} The token that was keyed out.
-  */
+ * Listens to any changes in the lifecycle managers logonChange model and calls the
+ * appropriate functions in gpii-app for notifying of the current user state
+ */
+gpii.app.logonChangeListener = function (that, model) {
+    if (model.inProgress === true) {
```
Comment (amb26) [inline src/app.js]: I like to write (model.inProgress ? gpii.app.displayWaitDialog : gpii.app.dismissWaitDialog)(that); but some might consider this over-condensed

File: src/app.js (hunk 1 above)
Comment (michelled) [inline src/app.js]: Addressed at 6b849bb4d0c62a247a5c8aeabd45505029213ad6

File: src/app.js (hunk 2 above)
Comment (michelled) [inline src/app.js]: Addressed at 6b849bb4d0c62a247a5c8aeabd45505029213ad6

File: package.json (hunk 3 above)
Comment (michelled) [inline package.json]: Addressed at 6b849bb4d0c62a247a5c8aeabd45505029213ad6

File: src/app.js (hunk 4 above)
Comment (michelled) [inline src/app.js]: Addressed at 6b849bb4d0c62a247a5c8aeabd45505029213ad6

File: src/app.js [hunk 5]
```diff
@@ -181,10 +224,60 @@ gpii.app.exit = function (that) {
...
+/**
+ * Shows the dialog on users screen with the message passed as parameter.
+ * Records the time it was shown in `dialogStartTime` which we need when
+ * dismissing it (checking whether it's been displayed for the minimum amount of time)
+ *
+ * @param that {Object} the app module
+ */
+gpii.app.displayWaitDialog = function (that) {
+    that.dialog.show();
+    // Hack to ensure it stays on top, even as the GPII autoconfiguration starts applications, etc., that might
```
Comment (michelled) [inline src/app.js]: @kaspermarkus I'm not sure this is doing what you intended. The on screen keyboard is placed on top of the wait dialog. Perhaps this isn't something we can control.

**Comment by michelled** [comment]
@javihernandez I updated my pull request and addressed the code review comments. Ready for another round of review.

**Pull request merged**